### ISSUE TITLE
Make Varien_Data_Collection::getFirstItem() associate the item to the collection

### ISSUE THREAD

**Issue by joshua-bn** [OpenMage/magento-lts#1363]
### Description (*)
https://github.com/OpenMage/magento-lts/blob/8bbc13bb622f767a08aca06b38b4007a8d2f5018/lib/Varien/Data/Collection.php#L276

You get the first item of the collection but it's not part of the collection. You have to add it yourself

### Expected behavior (*)
Before returning the new object, it adds it to the collection

### Benefits
One less thing to worry about

**Comment by joshua-bn** [comment]
Can someone retag this to developer experience? My mind blanked when I created it.

**Comment by colinmollenhour** [comment]
Returning an empty object when there is no first item is a bad design to begin with so there is a bit of a conflict here for best practice.. In my opinion this should not be changed as requested for the following reasons:

- a method starting with "get" should not be modifying anything (this rule is broken quite often, but still)
- assuming that the item should be added or with what method or the timing could all be incorrect assumptions (e.g. maybe they want to add it only after saving)
- auto-adding would be really hard to reverse so I think not adding it is actually better (it can easily be added but not so easily removed)

**Comment by joshua-bn** [comment]
I totally agree with you. I just did this because I spent 3 hours trying to figure out what was going on. So, I am just going to close this.
